src/build/env.py

```diff
@@ -1,312 +1,313 @@
 """
 Creates and manages isolated build environments.
 """
+
 import abc
 import functools
 import logging
 import os
 import platform
 import shutil
-import subprocess
 import sys
 import sysconfig
 import tempfile
 
-from types import TracebackType
-from typing import Callable, Iterable, List, Optional, Tuple, Type
-
-import build
-
+from typing import Dict, Iterable, Sequence, Tuple, Type
 
-if sys.version_info < (3, 8):
-    import importlib_metadata as metadata
-else:
-    from importlib import metadata
+from ._helpers import check_dependency, default_subprocess_runner
 
-try:
-    import virtualenv
-except ModuleNotFoundError:
-    virtualenv = None
 
+_logger = logging.getLogger(__name__)
 
-_logger = logging.getLogger('build.env')
+_cache = functools.partial(functools.lru_cache, maxsize=None)
 
 
 class IsolatedEnv(metaclass=abc.ABCMeta):
-    """Abstract base of isolated build environments, as required by the build project."""
+    """ABC for isolated build environments."""
 
-    @property
     @abc.abstractmethod
-    def executable(self) -> str:
-        """The executable of the isolated build environment."""
+    def __init__(self, isolated_env_manager: 'IsolatedEnvManager') -> None:
+        """
+        :param isolated_env_manager: The ``IsolatedEnvManager`` instance
+        """
         raise NotImplementedError
 
-    @property
     @abc.abstractmethod
-    def scripts_dir(self) -> str:
-        """The scripts directory of the isolated build environment."""
+    def create(self) -> None:
+        """
+        Create the isolated environment.
+
+        This method should be idempotent.
+        """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def install(self, requirements: Iterable[str]) -> None:
+    def install_packages(self, requirements: Iterable[str]) -> None:
```

Comment:
Any reason this needed to change from install to install packages? I don't see any other install methods here to need to specify? What else could you install?

Reply:
Methods without a noun can be assumed to operate on the class, so an unqualified `install` would install the environment.

Reply:
Interesting, I never made that assumption myself :thinking: